packages/mobile/src/components/premium-track-purchase-drawer/PremiumTrackPurchaseDrawer.tsx

```diff
@@ -126,61 +141,73 @@ export const PremiumTrackPurchaseDrawer = () => {
     !isUSDCEnabled
   )
     return null
-  const price = formatPrice(premiumConditions.usdc_purchase.price)
+
+  const price = premiumConditions.usdc_purchase.price
+  const purchaseSummaryValues = getPurchaseSummaryValues(price, currentBalance)
 
   return (
     <NativeDrawer
       drawerName={PREMIUM_TRACK_PURCHASE_MODAL_NAME}
       onClosed={handleClosed}
     >
-      <View style={styles.drawer}>
-        <View style={styles.titleContainer}>
-          <IconCart fill={neutralLight2} />
-          <Text style={styles.title}>{messages.title}</Text>
+      {isLoading ? (
+        <View style={styles.spinnerContainer}>
+          <LoadingSpinner />
```

Comment:
fwiw, I did not check with design on what the loading pattern should be for this flow.
It's possible we should be using shimmer states instead.

Reply:
good point, i don't anticipate this happening often but we can surface with design